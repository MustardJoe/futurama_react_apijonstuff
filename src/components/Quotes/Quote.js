import React from 'react';
import PropTypes from 'prop-types';

function Quote({ quoteObj }) {
  return (
    <section>
      <img src={quoteObj.image} />
      <p>{quoteObj.character}</p>
      <p>{quoteObj.quote}</p>
    </section>
  );
}

Quote.propTypes = {
  quoteObj: PropTypes.shape({
    image: PropTypes.string.isRequired,
    character: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
  }).isRequired
};

export default Quote;
