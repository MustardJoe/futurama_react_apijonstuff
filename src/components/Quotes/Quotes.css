.quoteListItem {
  width: 360px;
  overflow: hidden;
}