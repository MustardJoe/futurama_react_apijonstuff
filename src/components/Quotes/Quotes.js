import React from 'react';
import Quote from './Quote';
import PropTypes from 'prop-types';
import styles from './Quotes.css';

function Quotes({ quotes }) {
  const quoteList = quotes.map((quoteObj, i) => (
    <li className='quoteListItem' key={i}>
      <Quote quoteObj={quoteObj} />
    </li>
  ));

  return (
    <ul className={styles.Quotes}>
      {quoteList}
    </ul>
  );
}

Quotes.propTypes = {
  quotes: PropTypes.array.isRequired
};

export default Quotes;

