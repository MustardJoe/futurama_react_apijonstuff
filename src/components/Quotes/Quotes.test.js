import React from 'react';
import { shallow } from 'enzyme';
import Quotes from './Quotes';

describe('Quotes component test', () => {
  it('renders Quotes components, which is list of Quote comps', () => {
    const quotes = [{
      character: 'Fry',
      quote: 'What??',
      image: '/image'
    },
    {
      character: 'Fry',
      quote: 'Huh??',
      image: '/image'
    }];

    const wrapper = shallow(<Quotes quotes={quotes}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
