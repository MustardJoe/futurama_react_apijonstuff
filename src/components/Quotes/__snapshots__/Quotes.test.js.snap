// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component test renders Quotes components, which is list of Quote comps 1`] = `
<ul
  className="Quotes"
>
  <li
    key="0"
  >
    <Quote
      quote={
        Object {
          "character": "Fry",
          "image": "/image",
          "quote": "What??",
        }
      }
    />
  </li>
  <li
    key="1"
  >
    <Quote
      quote={
        Object {
          "character": "Fry",
          "image": "/image",
          "quote": "Huh??",
        }
      }
    />
  </li>
</ul>
`;
